actions/preview/action.yml

```diff
@@ -0,0 +1,97 @@
+name: 'Documentation Preview'
+description: 'Builds and publishes documentation to preview environment'
+
+branding:
+  icon: 'filter'
+  color: 'red'
+
+inputs:
+  strict:
+    description: 'Whether to fail the build if there are any warnings'
+    default: true
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+    - name: Create Deployment
+      uses: actions/github-script@v7
+      id: deployment
+      with:
+        result-encoding: string
+        script: |
+          const { owner, repo } = context.repo;
+          const branch = context.payload.pull_request
+            ? context.payload.pull_request.head.ref 
+            : context.ref.replace('refs/heads/', '');
+
+          const repoResponse = await github.rest.repos.get({
+            owner,
+            repo
+          })
+          const defaultBranch = repoResponse.data.default_branch;
+          const isDefaultBranch = branch === defaultBranch;
+
+          const environment = isDefaultBranch
+            ? `docs-preview-${defaultBranch}`
+            : `docs-preview-${context.issue.number}`;
+
+          const logUrl = `https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${context.runId}`;
+
+          const deployment = await github.rest.repos.createDeployment({
+              owner,
+              repo,
+              ref: branch,
+              environment,
+              auto_merge: false,
+              required_contexts: [],
+          })
+          await github.rest.repos.createDeploymentStatus({
+              deployment_id: deployment.data.id,
+              owner,
+              repo,
+              state: "in_progress",
+              log_url: logUrl,
+          })
+          return deployment.data.id
+
+    - name: Generate Path Prefix
+      id: path-prefix
+      shell: bash
+      run: |
+        if [[ -n "${{ github.event.pull_request }}" ]]; then
+          echo "result=/${GITHUB_REPOSITORY}/pull/${{ github.event.pull_request.number }}" >> $GITHUB_OUTPUT
+        else
+          echo "result=/${GITHUB_REPOSITORY}/${{ github.ref_name }}" >> $GITHUB_OUTPUT
+        fi
+
+    - name: Build public documentation
+      uses: elastic/docs-builder@main
+      continue-on-error: ${{ inputs.strict != 'true' }}
```

Comment:
Maybe add a comment that this a temporary? we want to be strict out of the box real soon :)

Reply:
Sounds good.

Thank you for the review. But I actually forgot to verify the preview of the default branch.

Will change the state to Draft again.